x-pack/filebeat/input/entityanalytics/provider/okta/okta.go

```diff
@@ -394,18 +407,155 @@ func (p *oktaInput) doFetch(ctx context.Context, state *stateStore, fullSync boo
 	// have a complete set from that timestamp.
 	query = url.Values{}
 	query.Add("search", fmt.Sprintf(`lastUpdated ge "%s" and status pr`, lastUpdated.Format(okta.ISO8601)))
-	state.next = query.Encode()
+	state.nextUsers = query.Encode()
 
 	p.logger.Debugf("received %d users from API", len(users))
 	return users, nil
 }
 
-func grow(u []*User, n int) []*User {
-	if len(u)+n <= cap(u) {
-		return u
+// doFetchDevices handles fetching device and associated user identities from Okta.
+// If fullSync is true, then any existing deltaLink will be ignored, forcing a full
+// synchronization from Okta.
+// Returns a set of modified devices by ID.
+func (p *oktaInput) doFetchDevices(ctx context.Context, state *stateStore, fullSync bool) ([]*Device, error) {
```

Comment:
Same question as before, why not just **fetchDevices** ?

Reply:
Same answer.